notebook/static/notebook/js/notebook.js

```diff
@@ -2876,10 +2876,13 @@ define([
                                 cell.output_area.trusted = true;
                             }
                         }
-                        nb.events.on('notebook_saved.Notebook', function () {
-                            window.location.reload();
+                        nb.contents.trust(nb.notebook_path)
+                            .then(function(res) {
+                                nb.events.trigger("trust_changed.Notebook", true);
+                                window.location.reload();
+                            }, function(err) {
+                                console.log(err);
                         });
-                        nb.save_notebook();
```

Comment:
Can you still trigger save here if the notebook is not read-only? That way we can avoid losing unsaved changes when the notebook is editable.

```python
var p;
if (nb.writable and nb.dirty) {
    p = nb.save_notebook();
} else {
    p = Promise.resolve();
}
p.then(function () {
    return nb.contents.trust...
```